compiler/rustc_lexer/src/lib.rs

```diff
@@ -413,7 +414,7 @@ impl Cursor<'_> {
             }
             _ => Unknown,
         };
-        Token::new(token_kind, self.len_consumed())
+        Token::new(token_kind, self.reset_len_consumed())
```

Comment:
Why do we reset the consumed length here now?

Reply:
So the next time `advance_token()` is called, the len_consumed is 0 again. 
That could also be done in the `tokenize()` function though.

Reply:
Is that not a change in behavior? I'm trying to follow where the length was previously reset, but not seeing it -- maybe you can give a hint?

Reply:
It's a few percent faster calling `reset_len_consumed()` here than in `tokenize()` though.

Reply:
But where was the previous call to reset it? I'm not seeing a modification of initial_len anywhere in the pre-PR code, so it seems like the previous code managed to keep track of the right length without resetting anything? What has changed?

Reply:
Previously, a new `Cursor` was created and thrown away every iteration. So re-creating did reset it.

(Sorry, GitHub didn't show your previous comment before I made my second one)

Reply:
`input = &input[token.len..]` previously changed the "initial len" used in the next iteration.

Reply:
Changed it back so that `reset_len_consumed()` is called in `tokenize()`. Rustc uses the other entry-point function anyway (oops) and it makes the `#[rustfmt::skip]` unneccessary.